src/intersection.class.ts

```diff
@@ -1,68 +1,91 @@
+ * @param T 
+ * @param A 
+ * @param B 
+ * @returns true if `T` is contained
+ */
+const isContainedInInterval = (T: Point, A: Point, B: Point) => {
+  const TA = new Point(T).subtract(A);
+  const TB = new Point(T).subtract(B);
+  return Math.sign(TA.x) !== Math.sign(TB.x) || Math.sign(TA.y) !== Math.sign(TB.y);
+}
+
+export class Intersection {
+
+  points: Point[]
+
+  status?: IntersectionType
+
+  constructor(status?: IntersectionType) {
     this.status = status;
     this.points = [];
   }
 
-  fabric.Intersection = Intersection;
-
-  fabric.Intersection.prototype = /** @lends fabric.Intersection.prototype */ {
-
-    constructor: Intersection,
+  /**
+   * 
+   * @param {Point} point 
+   * @returns 
+   */
+  contains(point) {
+    return this.points.some(p => p.eq(point));
+  }
 
-    /**
-     * Appends a point to intersection
-     * @param {Point} point
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoint: function (point) {
+  /**
+   * Appends a point to intersection
+   * @param {Point} point
+   * @return {Intersection} thisArg
+   * @chainable
+   */
+  appendPoint(point) {
+    if (!this.contains(point)) {
       this.points.push(point);
-      return this;
-    },
-
-    /**
-     * Appends points to intersection
-     * @param {Array} points
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoints: function (points) {
-      this.points = this.points.concat(points);
-      return this;
     }
-  };
+    return this;
+  }
 
   /**
-   * Checks if one line intersects another
-   * TODO: rename in intersectSegmentSegment
+   * Appends points to intersection
+   * @param {Point[]} points
+   * @return {Intersection} thisArg
+   * @chainable
+   */
+  appendPoints(points) {
+    this.points = this.points.concat(points.filter(point => {
+      return !this.contains(point);
+    }));
+    return this;
+  }
+
+  /**
+   * Checks if a line intersects another
    * @static
    * @param {Point} a1
    * @param {Point} a2
    * @param {Point} b1
    * @param {Point} b2
-   * @return {fabric.Intersection}
+   * @param {boolean} [aIinfinite=true] check intersection by passing `false`
+   * @param {boolean} [bIinfinite=true] check intersection by passing `false`
+   * @return {Intersection}
    */
-  fabric.Intersection.intersectLineLine = function (a1, a2, b1, b2) {
-    var result,
-        uaT = (b2.x - b1.x) * (a1.y - b1.y) - (b2.y - b1.y) * (a1.x - b1.x),
-        ubT = (a2.x - a1.x) * (a1.y - b1.y) - (a2.y - a1.y) * (a1.x - b1.x),
-        uB = (b2.y - b1.y) * (a2.x - a1.x) - (b2.x - b1.x) * (a2.y - a1.y);
+  static intersectLineLine(a1, a2, b1, b2, aIinfinite = true, bIinfinite = true) {
+    let result;
+    const uaT = (b2.x - b1.x) * (a1.y - b1.y) - (b2.y - b1.y) * (a1.x - b1.x),
+      ubT = (a2.x - a1.x) * (a1.y - b1.y) - (a2.y - a1.y) * (a1.x - b1.x),
+      uB = (b2.y - b1.y) * (a2.x - a1.x) - (b2.x - b1.x) * (a2.y - a1.y);
```

Comment:
who ever wrote this knows their algerba
I can't understand it exactly, just the main concept

Reply:
some matrix det/cross clever calculation
well done who ever you are

Reply:
Some of those can be probably grouped in smaller const, i m not sure the extra const are worth the subtraction saved.

Reply:
I thought of it too.
Wasn't sure how to name what...